### PR TITLE
Add a simple fps counter

### PR DESCRIPTION
**Pull request by amzn-rhhong**
https://user-images.githubusercontent.com/69218254/155468295-afa12e7d-e373-4ba1-87ac-940dd0c0ce7c.mp4

Signed-off-by: rhhong <rhhong@amazon.com>

### CODE REVIEW

**Review by amzn-jillich — COMMENTED** [review]
Isn't there something we can reuse here? Feels like there should be something part of the Atom widget. If I remember correctly, I have seen something for this - did you reach out to the Atom team?

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.h [hunk 1]
```diff
@@ -28,6 +29,20 @@ namespace AZ
 
 namespace EMStudio
 {
+    class FPSCounter
+    {
+    public:
+        void Step();
+        AZ::u32 GetFPS() { return m_lastFPS; }
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.h:36]: nit: We should add `const` here

**Review by hultonha — COMMENTED** [review]
I think @amzn-jillich's comment is valid, is the current FPS counter in the main viewport something we could reuse?

If that's not possible (if it's tied into the code where it's used), then this is a handy little construct you've created. Perhaps it could instead moved to `AzToolsFramework` or `AzFramework` so it could be reused in other viewports, not just EMFX?

**Comment by jeremyong-az** [comment]
The component you're looking for is the `AtomViewportDisplayInfoSystemComponent`. I recommend leveraging this as I'm about to add additional stats here in the future

**Comment by amzn-rhhong** [comment]
thanks for the suggestion @jeremyong-az @amzn-jillich @hultonha @gadams3 
I didn't use AtomViewportDisplayInfoSystemComponent for the reason that 1) it is tied to the default viewport 2) it seems to be a lot richer than a simple fps counter with graphic focus, I'm not sure if that is all needed or relevant for the animation viewport.

If we really have to use the atom fps counter, I can include that as part of the actorModule (inside o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\ActorModule.cpp, where most of the atom viewport code sits) and also modify the code a little bit so it can be flexible to target different viewport context.

**Review by hultonha — COMMENTED** [review]
This makes sense to me if this is the way Atom are doing it in the other viewport. Might be worth tagging @gadams3 to review as well as getting sign off from @jeremyong-az

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.h [hunk 2]
```diff
@@ -83,6 +88,10 @@ namespace EMStudio
         AZStd::shared_ptr<AzToolsFramework::ManipulatorManager> m_manipulatorManager;
         AZ::Transform m_mouseDownStartTransform;
 
+        // Atom performance metrics
+        QTimer m_metricsTimer;
+        AZStd::string m_fpsStr;
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.h:93]: uber nit: variable name could not use the abbreviation and be `m_fpsDisplay` or `m_fpsOutput` or something instead?

**Review by hultonha — APPROVED** [review]

**Comment by hultonha** [comment]
Relates to https://github.com/o3de/o3de/issues/7758

**Comment by amzn-jillich** [comment]
@amzn-rhhong is the animation viewport context and its name easily retrievable from the animation editor? Looks like these two places hard-code it to the default viewport:
```
        AZ::RPI::ViewportContextNotificationBus::Handler::BusConnect(
            AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContextName());
```
and
```
    AZ::RPI::ViewportContextPtr AtomViewportDisplayInfoSystemComponent::GetViewportContext() const
    {
        return AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext();
    }
```
Can we even have two of these components simultaneously? As they are named system component?

If yes, we might make an option to overwrite the viewport context and use that so that we get the additional statistics automatically and don't need to maintain that code?

**Comment by amzn-rhhong** [comment]
@amzn-jillich we are not using the AtomViewportDisplayInfoSystemComponent, but the PerformanceMonitorRequestBus as @gadams3 suggested. This is tied to PerformanceMonitorSystemComponent, which seems to be viewport independent.

**Review by amzn-jillich — APPROVED** [review]

**Pull request merged**